frontend/archive/src/main/java/org/pytorch/serve/archive/model/ModelConfig.java

```diff
@@ -0,0 +1,181 @@
+package org.pytorch.serve.archive.model;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ModelConfig {
+    private int minWorkers;
+    private int maxWorkers;
+    private int batchSize;
+    private int maxBatchDelay;
+    private int responseTimeout;
+    private DeviceType deviceType = DeviceType.NONE;
+    private List<Integer> deviceIds;
+    private int parallelLevel = 1;
+    private ParallelType parallelType = ParallelType.NONE;
+
+    public static ModelConfig build(Map<String, Object> yamlMap) {
```

Comment:
This seems like a schema, will this reduce flexibility for users trying to add their own configs are these configs from the YAML that matter to us. Curious why the frontend needs to be aware of of these in the first place